src/main/java/com/iota/iri/service/milestone/impl/MilestoneServiceImpl.java

```diff
@@ -65,9 +65,31 @@
     private MessageQ messageQ;
 
     /**
-     * Holds the config with important milestone specific settings.<br />
+     * Holds the coordinator address which is used to validate signatures.<br />
+     */
+    private Hash coordinatorAddress;
+
+    /**
+     * Holds the coordinator signature mode which is used to validate signatures.<br />
+     */
+    private SpongeFactory.Mode coordinatorSignatureMode;
+
+    /**
+     * Holds the coordinator security level is used to validate signatures.<br />
+     */
+    private int coordinatorSecurityLevel;
+
+    /**
+     * Holds the depth of Merkle tree used for coordinator signatures.<br />
+     */
+    private int numberOfKeysInMilestone;
+
+    /**
+     * Specifies if signatures should be validated.<br />
      */
-    private ConsensusConfig config;
+    private boolean skipValidation;
+
+    private int maxMilestoneIndex;
```

Comment:
Why did you decide to extract those from the config object and not just use the getters like before.
In my opinion this just saves lots of lines of code...

I think the nicer thing to do is to have `HashFactory.ADDRESS.create(config.getCoordinator())`.
I suppose `maxMilestoneIndex = 1 << config.getNumberOfKeysInMilestone();` can also be done as an automatically calculated config in my opinion.

The thing is that if we do this change here, we will have to do the same design in the rest of the code.

So what I am asking for you is either:
1. Explain why you think it is better to not use config getters. In this case then I insist that at least the fields will be `final`. According to DDD services should be stateless. I say that at the very least they should be immutable. (The `Hash` field won't be immutable, as you well know, but we will solve it)
2. Move everything back to the config and use getters (what I prefer). There is a reason why the setters are not visible

Reply:
It may be interesting to have a MilestoneService instance that can validate milestones coming from another coo than the one configured (wink, wink ...)

Reply:
The reason is that these parameters are fetched, constructed and calculated often. I think it's not efficient to do that, but I get the point of not storing state.

Anyways, I'm not sure I'll be available this week to make the change, if you find time to do it great, if not I'll update when I get to it.

Reply:
I wouldn't expect this to be the bottleneck in any scenario.
Doing a `get` should be really fast
I would expect that the HotSpots compiler to optimize such inefficiencies